src/components/application_manager/include/application_manager/resumption/resume_ctrl_impl.h

```diff
@@ -0,0 +1,491 @@
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_RESUMPTION_RESUME_CTRL_IMPL_H_
+#define SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_RESUMPTION_RESUME_CTRL_IMPL_H_
+
+#include <stdint.h>
+#include <vector>
+#include <map>
+#include <set>
+#include <list>
+
+#include "interfaces/HMI_API.h"
+#include "interfaces/HMI_API_schema.h"
+#include "interfaces/MOBILE_API_schema.h"
+#include "application_manager/event_engine/event_observer.h"
+#include "smart_objects/smart_object.h"
+#include "application_manager/application.h"
+#include "application_manager/resumption/resumption_data.h"
+#include "application_manager/resumption/resume_ctrl.h"
+#include "utils/timer.h"
+
+namespace resumption {
+
+class LastState;
+
+/**
+ * @brief Contains logic for storage/restore data of applications.
+ */
+
+class ResumeCtrlImpl : public ResumeCtrl,
+                       public application_manager::event_engine::EventObserver {
+ public:
+  /**
+   * @brief allows to create ResumeCtrlImpl object
+   */
+  ResumeCtrlImpl(application_manager::ApplicationManager& application_manager);
+
+  /**
+   * @brief allows to destroy ResumeCtrlImpl object
+   */
+  ~ResumeCtrlImpl();
+
+  /**
+   * @brief Event, that raised if application get resumption response from HMI
+   * @param event : event object, that contains smart_object with HMI message
+   */
+  void on_event(const app_mngr::event_engine::Event& event) OVERRIDE;
+
+  /**
+   * @brief Save all applications info to the file system
+   */
+  void SaveAllApplications();
+
+  /**
+   * @brief Save application persistent info for future resuming
+   * @param application is application witch need to be saved
+   */
+  void SaveApplication(app_mngr::ApplicationSharedPtr application) OVERRIDE;
+
+  /**
+   * @brief Set application HMI Level and ausio_state as saved
+   * @param application is application witch HMI Level is need to restore
+   * @return true if success, otherwise return false
+   */
+  bool RestoreAppHMIState(app_mngr::ApplicationSharedPtr application);
```

Comment:
@LevchenkoS Please add OVERRIDE here and in the other methods inherited from `ResumeCtrl`